### ISSUE TITLE
Add flag or global variable for the stacktrace of variable creation

### ISSUE THREAD

**Issue by lamblin** [Theano/Theano#3641]
By default, we strip everything from `theano.*` (except if `test` is in the filename), and keep the first line outside of Theano.
It would be better to be able to control that behaviour to include the full trace, especially for testing purposes.
See #3018.

**Comment by nouiz** [comment]
https://github.com/Theano/Theano/pull/4079 made the Theano flag theano.config.traceback.limit to determine the number of stack level to return to the user. This is always the number returned (except if there isn't as many stack level that exist).

So I think this issue is closed.

**Comment by lamblin** [comment]
Not really, the issue is also about being able to include the part from inside Theano, so we do not have to hardcode `tests`.
So we could reword this ticket to add a flag that makes `add_stack_trace` call `simple_extract_stack` with `skips=[]`, or maybe to control what is passed as `skips`.

**Comment by nouiz** [comment]
Why the current exception that skip tests don't do it? Framework above
Theano that wrap theano.function could want to add their own wrapper to
give a better user stack trace.

So yes, I agree, it can be useful.

**Comment by nouiz** [comment]
Currently, there is this Theano flag traceback.compile_limit. It default to 0. So we don't keep stack trace during compilation by default. If it is higher then 0, we have the side effect of not filtering the stack trace as we want to keep investigate inside Theano.

Do that fill the need from this issue?

**Comment by nouiz** [comment]
@lamblin do you think the current implementation in my previous comment fill the issue?

**Comment by lamblin** [comment]
Do you mean that if `traceback.compile_limit > 0`, the stack traces outside of the compilation will stop being filtered?
If so, that would be a temporary workaround, but I think we should separate both behaviours, which could be another part of this ticket.

**Comment by nouiz** [comment]
You understood correctly that current state. I don't understand why you
want to split. If we want the stack trace for nodes created by
optimization, then we want the stack trace inside Theano.

I don't know how this collide with the copy of the stack trace during
optimization. I think we need to not copy the stack trace when this flag is
used.
